ansible_base/lib/serializers/mixins.py

```diff
@@ -0,0 +1,24 @@
+import logging
+
+from rest_framework import serializers
+
+logger = logging.getLogger('ansible_base.lib.serializers.mixins')
+
+
+class ImmutableFieldsMixin(serializers.ModelSerializer):
+    def get_extra_kwargs(self):
+        kwargs = super().get_extra_kwargs()
+        immutable_fields = getattr(self.Meta, "immutable_fields", None)
+
+        # Make field read_only if instance already exists
+        if self.instance and immutable_fields:
+            for field in immutable_fields:
+                kwargs.setdefault(field, {})
+                kwargs[field]["read_only"] = True
```

Comment:
What does this do, if anything, to the options page? Would it be read_only = False on a list page and read_only = True on view page? Does this mess with the OpenAPI schema at all?

Reply:
For the options page, the fix is simple but not super clean.

It seems to involve either doing as directed in my previous comment, which leads to duplicated code, or overriding `get_serializer` to return an instanced one if action is `metadata`.

Reply:
I'm throwing in the towel on the OpenAPI spec, it should generate working API specs, just cannot seem to get the schema to change based on what context the serializer is generated.